rerun_cpp/src/rerun/recording_stream.hpp

```diff
@@ -9,6 +9,7 @@
 #include <vector>
 
 #include "as_components.hpp"
+#include "c/rerun.h"
```

Comment:
It's on me to not add a lint for that, but I really don't want to import all the public c methods into the user's namespace (this does just that). This is because it has a lot of free functions that are all kinda hard to use and quite unsafe. May easily confuse a newcomer looking into their autocomplete. So far `#include "c/rerun.h"` is only ever included in cpp files for that reason.
Can you please add this to `rerun_cpp\tests\header_checks.cpp` (ping me on slack if unclear how) and avoid this include?

.. also don't see why its needed in the first place :)